common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1633,7 +1633,12 @@ def rescore(self, only_if_higher=False):
         # rescoring should have no effect on attempts, so don't
         # need to increment here, or mark done.  Just save.
         self.set_state_from_lcp()
-        self.set_score(calculated_score)
+        old_percentage = float(orig_score.raw_earned) / float(orig_score.raw_possible)
+        new_percentage = float(calculated_score.raw_earned) / float(calculated_score.raw_possible)
+        # score should not be set if only_if_higher flag is set and
+        # old_percentage is greater than new percentage
+        if not (only_if_higher and old_percentage > new_percentage):
+            self.set_score(calculated_score)
```

Comment:
Thanks for investigating this issue.
I'm actually wondering whether `set_score` should even be called here.  I believe you can just delete the call to `set_score` altogether from this method - as it is then eventually called at https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/signals/handlers.py#L151-L155

That code in the grades-layer already does the before-after score comparison and it's better to have that logic in one place.

FYI @sanfordstudent - in case you have a different understanding of how this code-path is supposed to work.

Reply:
I think you're right!